### ISSUE TITLE
Upon saving a file Lapce changes the contents of the file

### ISSUE THREAD

**Issue by cauliflowew** [lapce/lapce#454]
I notice that sometimes when saving a file on macOS v12.3.1 (using precompiled binaries from the project's website), Lapce will edit the file's contents. An example below

https://user-images.githubusercontent.com/82113833/164992837-98c17fe0-858d-48a5-a991-ea6286bd849d.mov

:

**Comment by bugadani** [comment]
Interesting issue. What did exactly happen in this video?

Based on what you wrote, it looks like the following:

- you placed the cursor
- you saved the file
- `(` was replaced by `*`

Is this correct? Do you have the file open in another editor? Does the `*` version correspond to any existing code or an old revision of the same source??

**Comment by cauliflowew** [comment]
> * you placed the cursor
> * you saved the file
> * `(` was replaced by `*`

yes, this is correct, and no, the `*` doesn't really correspond to any previous iteration of this code, plus, if i delete the line entirely, save the file, then write the exact same line from scratch the same bug reappears

**Comment by cauliflowew** [comment]
> * you placed the cursor
> * you saved the file
> * `(` was replaced by `*`

yes, this is correct, and i do have another text editor with this file open, however, after i closed both the second editor and Lapce, and reopened Lapce, the problem persists

the way i work around this, is by doing all the normal text editing in Lapce, then, when i encounter such a bug, i just go a second editor, fix the text there, save the file from the second editor, and then run

> Does the * version correspond to any existing code or an old revision of the same source??

maybe, i did change this line and this entire file many times, additionally, when i just open a new file in Lapce, and copy text in there and save, this error does not occur
